waterloo-rocketry/topside
# Add support for building standalone executables

This commit adds a build.sh script that uses cx_Freeze to create a directory with an executable and the libraries it needs. We should be able to just zip this directory up and send it to the user. A script to create this package will be added in a future PR.

On my machine, the folder created is currently over 900 MB. Ideally, we would want to reduce that size somewhat; I've briefly looked into some options for excluding dependencies that we don't specifically need, but that will also be addressed in future PRs.

Theoretically, this script _should_ work on all platforms, but it's only been tested on Windows so far.

Testing instructions:
1. Install `cx-freeze` with pip if you don't have it already (it's in `requirements.txt` as well)
2. Run `build.sh`. You should see a lot of output fly past and eventually "Successfully built standalone executable"
3. If everything worked fine, there should now be a `build` directory in the repo root. Inside this folder, assuming you're on Windows, there will be a subdirectory starting with `exe.win`. Inside _that_ directory, you should see `OperationsSimulator.exe`.
4. Run `OperationsSimulator.exe`. After a few seconds, the program should start. It will likely be a bit slow to launch the first time you start it, but should start within a second on subsequent attempts.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/waterloo-rocketry/operations-simulator/2)
